Raven.Database/Raft/ClusterManager.cs

```diff
@@ -234,6 +290,34 @@ public void Dispose()
 
             aggregator.ThrowIfNeeded();
         }
+
+        private bool disableReplicationStateChecks;
+        private int disableReplicationStateChecksLocker;
+        public void HandleClusterConfigurationChanged(ClusterConfiguration configuration)
+        {
+            if (Interlocked.CompareExchange(ref disableReplicationStateChecksLocker, 1, 0) == 1)
+                return;
+            try
+            {
+                //same value nothing to do
+                if (configuration.DisableReplicationStateChecks == disableReplicationStateChecks)
+                    return;
+                if (configuration.DisableReplicationStateChecks)
+                {
+                    Engine.StateChanged -= OnRaftEngineStateChanged;
+                    SaflyDisposeOfTimer();
+                }
+                else
+                {
+                    Engine.StateChanged += OnRaftEngineStateChanged;
```

Comment:
shouldn't you also start the timer here?
If I changed it to false, then to true?